### PR TITLE
fix: Change render order of various decorations: icons > names > bars > selection

### PR DESCRIPTION
**Pull request by moreginger**

### CODE REVIEW

**Comment by moreginger** [comment]
Before:

image

After:

image

File: LuaUI/Widgets/gui_selectedunits_gl4.lua [hunk 1]
```diff
@@ -7,7 +7,7 @@ function widget:GetInfo()
 		license = "GNU GPL, v2 or later",
 		-- Somewhere between layer -40 and -30 GetUnitUnderCursor starts
 		-- returning nil before GetUnitsInSelectionBox includes that unit.
-		layer = -30,
+		layer = -8, -- Draw below health bars and selection.
```
Comment (sprunk) [inline LuaUI/Widgets/gui_selectedunits_gl4.lua:10]: selection is this one
```suggestion
		layer = -8, -- Draw below health bars, icons, and nametags
```

File: LuaUI/Widgets/gui_name_tags.lua [hunk 2]
```diff
@@ -5,7 +5,7 @@ function widget:GetInfo()
 		author  = "fiendicus_prime",
 		date    = "2023-11-29",
 		license = "GNU GPL, v2 or later",
-		layer   = -9,
+		layer   = -11,
```
Comment (sprunk) [inline LuaUI/Widgets/gui_name_tags.lua:8]: -- above healthbars and selections, but below icons

Comment (sprunk) [inline LuaUI/Widgets/gui_name_tags.lua:8]:
```suggestion
		layer   = -11, -- above selections and healthbars, but below icons
```

File: LuaUI/Widgets/gui_selectedunits_gl4.lua (hunk 1 above)
Comment (moreginger) [reply LuaUI/Widgets/gui_selectedunits_gl4.lua:10]: Oops. Is it a good idea to enumerate all the options (which isn't a complete list here anyway). Or I suppose we could import constants?

**Review by sprunk — APPROVED** [review]

**Pull request merged**